src/main/java/org/sagebionetworks/web/client/widget/upload/FileValidator.java

```diff
@@ -0,0 +1,7 @@
+package org.sagebionetworks.web.client.widget.upload;
+
+public interface FileValidator {
+	
+	boolean isValid(String fileName);
```

Comment:
You should try to use the content type coming from the browser.  Then, if not set/recognized, then fall back to the filename (extension).  So this should take the FileMetadata and be reworked.